optaplanner-core/src/main/java/org/optaplanner/core/api/domain/variable/VariableListener.java

```diff
@@ -29,8 +28,11 @@
  * <p>
  * Each {@link ScoreDirector} has a different {@link VariableListener} instance, so it can be stateful.
  * If it is stateful, it must implement {@link StatefulVariableListener}.
+ * <p>
+ * Implementation note: if/when any new method is added to this interface, make sure that
+ * {@link SupplyVariableListener} overrides it and delegates to it.
  */
-public interface VariableListener<Solution_, Entity_> extends Supply {
+public interface VariableListener<Solution_, Entity_> {
```

Comment:
`Supply` is not a public API, so I removed it. Which forced me to do this:
https://github.com/kiegroup/optaplanner/pull/967/files#diff-1915332f290b70fd6c30cb1b442821c7b9fd64423386942e3e73c817707b5205R77